python/cvmfs/repository.py

```diff
@@ -197,39 +196,230 @@ def _pop_catalog(self):
         return wrapper.get_catalog()
 
 
+class Resource(file):
```

Comment:
Okay, I think with the `Cache::transaction()` and `Cache::commit()` its fine to use this class in an RAII sense. I'd just rename it to `CacheTransactionFile` or something. Resource is too generic for this. It is tightly coupled to the cache and should also reflect that in its name.